source_middleware.go

```diff
@@ -14,17 +14,42 @@
 
 package sdk
 
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"sync"
+
+	"github.com/conduitio/conduit-commons/config"
+	"github.com/conduitio/conduit-commons/opencdc"
+	"github.com/conduitio/conduit-commons/rabin"
+	"github.com/conduitio/conduit-commons/schema"
+	"github.com/conduitio/conduit-connector-sdk/internal"
+	sdkschema "github.com/conduitio/conduit-connector-sdk/schema"
+)
+
 // SourceMiddleware wraps a Source and adds functionality to it.
 type SourceMiddleware interface {
 	Wrap(Source) Source
 }
 
+// SourceMiddlewareOption can be used to change the behavior of the default source
+// middleware created with DefaultSourceMiddleware.
+type SourceMiddlewareOption func(SourceMiddleware)
+
 // DefaultSourceMiddleware returns a slice of middleware that should be added to
 // all sources unless there's a good reason not to.
-func DefaultSourceMiddleware() []SourceMiddleware {
-	return []SourceMiddleware{
-		// TODO add default middleware (e.g. rate limit)
+func DefaultSourceMiddleware(opts ...SourceMiddlewareOption) []SourceMiddleware {
```

Comment:
This makes it possible to configure default middleware. How about the case where we want to include only some of the middleware with `sdk.SourceWithMiddleware(&fooconnector.Source{}, sdk.BarMiddleware{})`?

Reply:
In that case you can set the defaults directly in the middleware struct, no need for options in that case:
```
sdk.SourceWithMiddleware(
    &fooconnector.Source{},
    sdk.BarMiddleware{
        Config: sdk.BarMiddlewareConfig{
            ParamFoo: "foo",
            ParamBar: 123,
        },
    },
)       
```

In other words, the options only play a role if you want to change the config of the default middleware, but retain the full list of the default middleware and benefit from new middleware if it gets added in future releases.

Alternatively, we could rename the options to `DefaultSourceMiddlewareOption`..?

Reply:
Ah, right, good old struct init, no need for anything more "clever" than that!

As for `DefaultSourceMiddlewareOption`, it sounds good to me, but I'm strongly on this or either side.:)

Reply:
I'll leave it as it is then, to not make it more verbose than it has to be 🙃